### DISCUSSION TITLE
[Feature] Dedicated support for basic-auth (Android/Exoplayer does not support BasicAuth in URL)

### DISCUSSION THREAD

**Discussion by poisonnuke** [immich-app/immich#12985] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

There are two options for securing immich when running on a public server:
VPN or BasicAuth Proxy.

as basic-auth is way easier to implement on mobile and server, it has a lot of advantages over the VPN solution and is therefore the option many users are choosing. 

There is only one issue: even though almost everything works when using the URL scheme for BasicAuth (user:password@TLD...) Android itself does not support URLs with that format.

Those two bug-reports go into the exact details, why Exoplayer and Flutter cannot play BasicAuth protected URLs by default and Exoplayer-team cannot solve the issue and the Flutter-team does not want to). But when Basic Auth is supplied seperatly to Flutter, BasicAuth is officially supported. 

https://github.com/flutter/flutter/issues/150411

https://github.com/androidx/media/issues/1627

Therefore the easiest solution would be to use provide a dedicated config option for immich-mobile app to support Basic Auth.

This would also improve security as now the password could be stored in an encrypted fashion and not cleartext in the URL.

This would ultimately solve the issue a lot of immich-users are facing, that videos cannot be played on the Android-app. 

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#5553

**Reply by poisonnuke** [reply]
that bug-report is incorrect and is ontop confusing. Noone understands it anymore. 

Please close the linked bug-report and use this request here. I have researched all options and provided a clean report that is exact and correct and provides the needed solution (which is a dedicated option for basic auth that then can be handed to the flutter-player)

**Reply by bo0tzz** [reply]
> that bug-report is incorrect and is ontop confusing. Noone understands it anymore.

What do you base this on? The issue seems pretty clear, and a fix is being explored. Further, that bug covers both mTLS and basic auth while your request here only covers the latter.

**Reply by poisonnuke** [reply]
ok there is at least so much confusion in the bug-report that the problem is not clear anymore. This request here is well researched and provides all details needed (way more details then any of you previously had!) to solve one of the problems. 

I did not see mTLS though, but it still makes sense to have two seperate stories to prevent any mixup or misunderstanding (which cleary happend here). So the logical step is to still close the bug-report and open an mTLS story, providing all the details for the mTLS issue. And its actually very likely that both stories need to be solved seperatly as the BA problem requires different steps to be taken.

**Reply by poisonnuke** [reply]
can we at least sort this out? Your answer is pretty disrespectful as I have invested weeks of time to debug through the entire media-stack to figure out what the bug is in the first place, no one did that before me. 

And your answer suggest that everything I did was worthless.

Looks like contribution is not welcome here?

**Reply by jrasm91** [reply]
Contributions are welcome but we also want to be effective and efficient. There is another discussion which also describes the shortcomings of the asset viewers in flutter, which has existed and has more discussion already. I think it would make more sense to add this information to that discussion instead.

**Reply by poisonnuke** [reply]
btw there is one caviat though, the other discussion is explicitly titled about self signed certs, whereas this issue here applies to all kinds of HTTP independently from the encryption. 

you really should create two issues, because I dont thinkt it'll help anyone if we use the other issue with all that back and forth in the discussion. In the end we risk solving only one of the issues because the developer taking care of it didnt see this one here. Its just not agile and correct if you leave the other discussion as ambigiuos as it is right now.
